### DISCUSSION TITLE
XYChart remove inner padding between the chart and the axis

### DISCUSSION THREAD

**Discussion by matecsigi** [airbnb/visx#1700] [Q&A] [answered]
Hi,

I have been using visx for a few months now and it has been great working with this library.
There is a padding question that I didn't find an answer to: Is there a way to remove the padding between the chart and the vertical axis (highlighted in red)?
<img width="488" alt="Screenshot 2023-05-04 at 15 55 27" src="https://user-images.githubusercontent.com/30734279/236211343-4e2876ad-9ba7-4c62-9975-fa2d746b2775.png">

Codesandbox link: https://codesandbox.io/s/withered-sun-1rfyyz?file=/src/App.tsx

Thanks!

**Comment by williaster** [comment, marked as answer]
Hey @matecsigi thanks for checkout out `visx`. This spacing comes from the `band` type scale which you are using for the x-axis. this is often used for bar charts where each point has a width – and in this case the line point is in the middle of the band (the [d3-scale docs](https://github.com/d3/d3-scale#band-scales) have a good visual explanation of this)

<img width="755" alt="image" src="https://user-images.githubusercontent.com/4496521/236274586-8152814b-0000-411f-9203-e186ad076a36.png">


Here's an update with a time scale which corrects the problem https://codesandbox.io/s/fancy-mountain-d7u9v4?file=/src/App.tsx

<img width="525" alt="image" src="https://user-images.githubusercontent.com/4496521/236274426-61579ef5-360b-4861-aecb-a068f444522b.png">

**Reply by matecsigi** [reply]
Thank you!
